doc/devguide.md

```diff
@@ -0,0 +1,130 @@
+# Azure IoT SDK Developer Documentation
+
+## 1. Dev machine prerequisites
+
+Before starting development on the Azure IoT SDK for C# you will need to install a few frameworks and tools. Please follow the instructions within [devbox_setup](devbox_setup.md).
+
+## 2. Coding style
+
+Please read and apply our [coding style](coding-style.md) when proposing PRs against the Azure repository.
+
+## 3. Cloning the repository
+
+`git clone https://github.com/Azure/azure-iot-sdk-csharp.git`
+
+## 4. Building the repository
+
+### Building the main repository
+
+* Windows: `build.cmd`
+* Linux: `./build.sh`
+
+For more information on running more complex scenarios (like e2e or stress testing) please add the `-h` argument to the build command.
+
+### Building a single project
+
+During development you can use Visual Studio by loading the `azureiot.sln` solution.
+If you are using a different editor, you can use the `dotnet` CLI to build (in Windows, Linux or OSX). Here are a few commands. For a complete list see the [.NET Core CLI tools](https://docs.microsoft.com/en-us/dotnet/core/tools/?tabs=netcore2x)
+
+* Build a library csproj: `dotnet build`
+* Run tests from a test csproj: `dotnet test`
+* Run an application from a csproj (e.g. a sample): `dotnet run`
+
+### ARM32 builds
+
+Our NuGet-packaged SDK can be used certain ARM32-based devices. Since the .NET SDKs are not available on this platform, you will need to cross-compile your application.
+
+Instructions are available here: https://github.com/dotnet/core/blob/master/samples/RaspberryPiInstructions.md
+
+### Xamarin, UWP, Windows IoT
+
+Please follow the [devbox-setup][devbox_setup.md] Optional prerequisites for Xamarin before trying to build. To build and test, run the following Jenkins script from the root of the repository:
+
+`jenkins\windows_csharp_xamarin.cmd`
+
+### .NET Micro Framework
+
+Please follow the [devbox-setup][devbox_setup.md] Optional prerequisites for Xamarin before trying to build. To build and test, run the following Jenkins script from the root of the repository:
```

Comment:
Please follow the [devbox-setup][devbox_setup.md] Optional prerequisites for _.NET Micro Framework_ before trying to build